src/React/Resources/assets/src/render_controller.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+'use strict';
+
+import React, {ReactElement} from 'react';
+import { Controller } from '@hotwired/stimulus';
+
+export default class extends Controller {
+    readonly componentValue: string;
+    readonly propsValue: object;
+
+    static values = {
+        component: String,
+        props: Object,
+    };
+
+    connect() {
+        this._dispatchEvent('react:connect', { component: this.componentValue, props: this.propsValue });
+
+        // Use a timeout to avoid mounting and demounting right after
+        if ((this.element as any).timeout) {
+            clearTimeout((this.element as any).timeout);
+        }
+
+        (this.element as any).timeout = setTimeout(() => {
+            const component = window.resolveReactComponent(this.componentValue);
+            this._renderReactElement(React.createElement(component, this.propsValue, null));
+
+            this._dispatchEvent('react:mount', { component: this.componentValue, props: this.propsValue });
```

Comment:
Pass the object too?

```suggestion
            this._dispatchEvent('react:mount', { componentName: this.componentValue, component: component, props: this.propsValue });
```